var/spack/repos/builtin/packages/qgis/pyqt5.patch

```diff
@@ -0,0 +1,25 @@
+diff --git a/cmake/FindPyQt5.py b/cmake/FindPyQt5.py
+index 6a55a0f801..d6eda1fa1e 100644
+--- a/cmake/FindPyQt5.py
++++ b/cmake/FindPyQt5.py
+@@ -40,6 +40,7 @@ except ImportError:
+     import sys
+     cfg = sipconfig.Configuration()
+     sip_dir = cfg.default_sip_dir
++    pyqt_prefix = os.sep.join(PyQt5.QtCore.__file__.split(os.sep)[0:-5])
```

Comment:
os.path.join

Reply:
```
In [1]: import PyQt5.QtCore

In [7]: PyQt5.QtCore.__file__.split(os.sep)[0:-2]
Out[7]: ['', 'usr', 'lib', 'python2.7', 'dist-packages']

In [9]: os.path.join(PyQt5.QtCore.__file__.split(os.sep)[0:-2])
Out[9]: ['', 'usr', 'lib', 'python2.7', 'dist-packages']

In [10]: os.path.join(*PyQt5.QtCore.__file__.split(os.sep)[0:-2])
Out[10]: 'usr/lib/python2.7/dist-packages'

In [12]: os.sep.join(PyQt5.QtCore.__file__.split(os.sep)[0:-2])
Out[12]: '/usr/lib/python2.7/dist-packages'
```
Above note the missing leading `/` when we do `os.path.join`. So `os.sep.join` is the right choice it seems.

Reply:
or the other option is:
```
In [6]: os.path.join(os.sep,*PyQt5.QtCore.__file__.split(os.sep)[0:-2])
Out[6]: '/usr/lib/python2.7/dist-packages'
```